src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.ILLink.targets

```diff
@@ -29,31 +29,57 @@ Copyright (c) .NET Foundation. All rights reserved.
     included in the generated deps.json.
     ============================================================
     -->
+  <UsingTask TaskName="FindNativeDeps" AssemblyFile="$(ILLinkTasksAssembly)" />
   <Target Name="_ILLink"
           Condition=" '$(PublishTrimmed)' == 'true' And
                       '$(_TargetFrameworkVersionWithoutV)' >= '3.0' And '$(TargetFrameworkIdentifier)' == '.NETCoreApp' "
           DependsOnTargets="_RunILLink">
 
+    <ItemGroup>
+      <_LinkedManagedAssemblies Include="@(_LinkedManagedAssembliesCandidates)" Condition="Exists('%(Identity)')" />
+    </ItemGroup>
+
+    <!-- Filter out native assets no longer needed after linking. -->
+    <ItemGroup Condition=" '$(TrimmerKeepNativeAssets)' != 'true' ">
+      <_NativeAssetsToAlwaysKeep Include="coreclr.dll;libcoreclr.dylib;libcoreclr.so" />
+      <_NativeAssetsToAlwaysKeep Include="clrjit.dll;libclrjit.dylib;libclrjit.so" />
+      <_NativeAssetsToAlwaysKeep Include="hostfxr.dll;libhostfxr.dylib;libhostfxr.so" />
+      <_NativeAssetsToAlwaysKeep Include="hostpolicy.dll;libhostpolicy.dylib;libhostpolicy.so" />
```

Comment:
What about the SOS/DAC binaries?

Reply:
Also, when @swaroop-sridhar and I were trying a few tests last week, we noticed that System.Private.CoreLib.dll was dropped. We tried a helloworld (references System.Runtime.dll, which references S.P.Corelib.dll), but neither dependencies seemed to get copied (maybe that's a different bug that we can fix in a different PR)

Reply:
I believe that's fixed with https://github.com/dotnet/sdk/pull/3156.

Reply:
Thanks for confirming